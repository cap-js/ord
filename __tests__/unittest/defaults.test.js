const defaults = require('../../lib/defaults');

describe('defaults', () => {
    describe('$schema', () => {
        it('should return default value', () => {
            expect(defaults.$schema).toMatchSnapshot();
        });
    });

    describe('openResourceDiscovery', () => {
        it('should return default value', () => {
            expect(defaults.openResourceDiscovery).toMatchSnapshot();
        });
    });

    describe('policyLevel', () => {
        it('should return default value', () => {
            expect(defaults.policyLevel).toMatchSnapshot();
        });
    });

    describe('description', () => {
        it('should return default value', () => {
            expect(defaults.description).toMatchSnapshot();
        });
    });

    describe('products', () => {
        const testProductsName = 'My Product';
        it('should return default value', () => {
            expect(defaults.products(testProductsName)).toMatchSnapshot()
        });
    });

    describe('groupTypeId', () => {
        it('should return default value', () => {
            expect(defaults.groupTypeId).toMatchSnapshot()
        });
    });

    describe('packages', () => {
        const testGetPackageDataName = 'My Package';
        const testGetPackageOrdNamespace = 'customer.sample';
        it('should return default value if policyLevel contains sap', () => {
            const testPolicyLevel = 'sap:policy';

            expect(defaults.packages(testGetPackageDataName, testPolicyLevel, testGetPackageOrdNamespace)).toMatchSnapshot();
        });

        it('should return default value if policyLevel does not contain sap', () => {
            const testPolicyLevel = 'policy';

            expect(defaults.packages(testGetPackageDataName, testPolicyLevel, testGetPackageOrdNamespace)).toMatchSnapshot();
        });
    });

    describe('consumptionBundles', () => {
        const testConsumptionBundlesName = 'My Consumption Bundle';
        it('should return default value', () => {
            expect(defaults.consumptionBundles(testConsumptionBundlesName)).toMatchSnapshot();
        });
    });
    describe('baseTemplate', () => {
        it('should return default value', () => {
            expect(defaults.baseTemplate).toMatchSnapshot();
        });
    });
});
