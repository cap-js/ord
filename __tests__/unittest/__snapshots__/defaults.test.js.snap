// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaults $schema should return default value 1`] = `"https://open-resource-discovery.github.io/specification/spec-v1/interfaces/Document.schema.json"`;

exports[`defaults baseTemplate should return default value 1`] = `
{
  "openResourceDiscoveryV1": {
    "documents": [
      {
        "accessStrategies": [
          {
            "type": "open",
          },
        ],
        "url": "/ord/v1/documents/ord-document",
      },
    ],
  },
}
`;

exports[`defaults consumptionBundles should return default value 1`] = `
[
  {
    "description": "This Consumption Bundle contains all resources of the reference app which are unprotected and do not require authentication",
    "lastUpdate": "2024-06-20T14:04:01+01:00",
    "ordId": "sapxref:consumptionBundle:noAuth:v1",
    "shortDescription": "If we have another protected API then it will be another object",
    "title": "Unprotected resources",
    "version": "1.0.0",
  },
]
`;

exports[`defaults description should return default value 1`] = `"this is an application description"`;

exports[`defaults groupTypeId should return default value 1`] = `"sap.cds:service"`;

exports[`defaults openResourceDiscovery should return default value 1`] = `"1.9"`;

exports[`defaults packages should return default value if policyLevel contains sap 1`] = `
[
  {
    "description": "This package contains public APIs for My Package.",
    "ordId": "customer.sample:package:MyPackage-api:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing public APIs",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains internal APIs for My Package.",
    "ordId": "customer.sample:package:MyPackage-api-internal:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing internal APIs",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains private APIs for My Package.",
    "ordId": "customer.sample:package:MyPackage-api-private:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing private APIs",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains public Events for My Package.",
    "ordId": "customer.sample:package:MyPackage-event:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing public Events",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains internal Events for My Package.",
    "ordId": "customer.sample:package:MyPackage-event-internal:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing internal Events",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains private Events for My Package.",
    "ordId": "customer.sample:package:MyPackage-event-private:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing private Events",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains public Consumption Bundles for My Package.",
    "ordId": "customer.sample:package:MyPackage-consumptionBundle:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing public Consumption Bundles",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains internal Consumption Bundles for My Package.",
    "ordId": "customer.sample:package:MyPackage-consumptionBundle-internal:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing internal Consumption Bundles",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains private Consumption Bundles for My Package.",
    "ordId": "customer.sample:package:MyPackage-consumptionBundle-private:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing private Consumption Bundles",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains public Integration Dependencies for My Package.",
    "ordId": "customer.sample:package:MyPackage-integrationDependency:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing public Integration Dependencies",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains internal Integration Dependencies for My Package.",
    "ordId": "customer.sample:package:MyPackage-integrationDependency-internal:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing internal Integration Dependencies",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains private Integration Dependencies for My Package.",
    "ordId": "customer.sample:package:MyPackage-integrationDependency-private:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing private Integration Dependencies",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains public Entity Types for My Package.",
    "ordId": "customer.sample:package:MyPackage-entityType:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing public Entity Types",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains internal Entity Types for My Package.",
    "ordId": "customer.sample:package:MyPackage-entityType-internal:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing internal Entity Types",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains private Entity Types for My Package.",
    "ordId": "customer.sample:package:MyPackage-entityType-private:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing private Entity Types",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains public Data Products for My Package.",
    "ordId": "customer.sample:package:MyPackage-dataProduct:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing public Data Products",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains internal Data Products for My Package.",
    "ordId": "customer.sample:package:MyPackage-dataProduct-internal:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing internal Data Products",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
  {
    "description": "This package contains private Data Products for My Package.",
    "ordId": "customer.sample:package:MyPackage-dataProduct-private:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing private Data Products",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
]
`;

exports[`defaults packages should return default value if policyLevel does not contain sap 1`] = `
[
  {
    "description": "This package contains public General for My Package.",
    "ordId": "customer.sample:package:MyPackage:v1",
    "partOfProducts": [
      "customer:product:My.Package:",
    ],
    "shortDescription": "Package containing public General",
    "title": "My Package",
    "vendor": "customer:vendor:Customer:",
    "version": "1.0.0",
  },
]
`;

exports[`defaults policyLevel should return default value 1`] = `"none"`;

exports[`defaults products should return default value 1`] = `
[
  {
    "ordId": "customer:product:My.Product:",
    "shortDescription": "Short description of My Product",
    "title": "My Product",
    "vendor": "customer:vendor:customer:",
  },
]
`;
